lib/chef/provider/windows_task.rb

```diff
@@ -52,40 +52,58 @@ def set_current_resource(task_hash)
       end
 
-      def action_create
-        if @current_resource.exists && !(task_need_update? || @new_resource.force)
-          Chef::Log.info "#{@new_resource} task already exists - nothing to do"
-        else
-          options = {}
-          options["F"] = "" if @new_resource.force || task_need_update?
-          options["SC"] = schedule
-          options["MO"] = @new_resource.frequency_modifier if frequency_modifier_allowed
-          options["I"]  = @new_resource.idle_time unless @new_resource.idle_time.nil?
-          options["SD"] = @new_resource.start_day unless @new_resource.start_day.nil?
-          options["ST"] = @new_resource.start_time unless @new_resource.start_time.nil?
-          options["TR"] = @new_resource.command
-          options["RU"] = @new_resource.user
-          options["RP"] = @new_resource.password if use_password?
-          options["RL"] = "HIGHEST" if @new_resource.run_level == :highest
-          options["IT"] = "" if @new_resource.interactive_enabled
-          options["D"] = @new_resource.day if @new_resource.day
-          options["M"] = @new_resource.months unless @new_resource.months.nil?
-
-          run_schtasks "CREATE", options
-          xml_options = []
-          xml_options << "cwd" if new_resource.cwd
-          xml_options << "random_delay" if new_resource.random_delay
-          xml_options << "execution_time_limit" if new_resource.execution_time_limit
-          update_task_xml(xml_options) unless xml_options.empty?
+      # This method checks if task and command attributes exist since those two are mandatory attributes to create a schedules task.
+      def basic_validation
+        validate = []
+        validate << "Command" if @new_resource.command.nil? || @new_resource.command.empty?
+        validate << "Task Name" if @new_resource.task_name.nil? || @new_resource.task_name.empty?
+        return true if validate.empty?
+        Chef::Log.error "Value for '#{validate.join(', ')}' option cannot be empty"
+        false
+      end
 
-          new_resource.updated_by_last_action true
-          Chef::Log.info "#{@new_resource} task created"
+      def action_create
+        if @current_resource.exists
+          if !(task_need_update? || @new_resource.force)
+            Chef::Log.info "#{@new_resource} task already exists - nothing to do"
+            return
+          end
+          # To merge current resource and new resource attributes
+          temp_arry = %w{ command user run_level cwd frequency_modifier frequency idle_time random_delay execution_time_limit start_day start_time }
+          temp_arry.each do |attribute|
+            new_resource_attribute = @new_resource.send(attribute)
+            current_resource_attribute = @current_resource.send(attribute)
+            @new_resource.send("#{attribute}=", current_resource_attribute ) if current_resource_attribute && new_resource_attribute.nil?
+          end
         end
+        return unless basic_validation
```

Comment:
By raise an error I meant raise an exception that will halt the execution.  While that line will emit an error to the log, execute will continue as if nothing was wrong unless I'm missing something?